src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SessionState/RemoteSession/RemoteAppSessionStateExtensions.cs

```diff
@@ -13,10 +14,20 @@ public static ISystemWebAdapterRemoteClientAppBuilder AddSessionClient(this ISys
     {
         ArgumentNullException.ThrowIfNull(builder);
 
-        builder.Services.AddTransient<ISessionManager, RemoteAppSessionStateManager>();
+        builder.Services.AddTransient<DoubleConnectionRemoteAppSessionManager>();
+        builder.Services.AddTransient<SingleConnectionWriteableRemoteAppSessionStateManager>();
+        builder.Services.AddTransient<ISessionManager, RemoteAppSessionDispatcher>();
 
         builder.Services.AddOptions<RemoteAppSessionStateClientOptions>()
             .Configure(configure ?? (_ => { }))
+            .PostConfigure<IOptions<RemoteAppClientOptions>>((options, remote) =>
+            {
+                // The single connection remote app session client requires https to work so if that's not the case, we'll disable it
+                if (!string.Equals(remote.Value.RemoteAppUrl.Scheme, "https", StringComparison.OrdinalIgnoreCase))
```

Comment:
Might be worth logging a warning here to let the user know why single connection isn't being used.